src/dash-table/components/FilterFactory.tsx

```diff
@@ -80,6 +95,11 @@ export default class FilterFactory {
                         operator,
                         setFilter
                     )}
+                    // Running into TypeScript binding issues with many parameters..
+                    // bind with no more than 4 params each time.. sigh..
+                    toggleFilterOptions={this.onToggleChange
+                        .bind(this, column, map, operator, setFilter)
+                        .bind(this, toggleFilterOptions, ast && ast.query)}
```

Comment:
So TypeScript defaults to a generic `...args: <type of first arg>[]` signature when there's more than 4 arguments apart from context. Breaking it up just so I don't get an error. Annoying limitation.

Reply:
Only on `bind`. The context is part of the bind signature so yes. R.partial suffers from the same limitations, see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ramda/index.d.ts#L1369-L1378. The problem isn't bind or R.partial per say but the type definitions.

Reply:
For what it's worth here is the typedef for bind

```
    /**
     * For a given function, creates a bound function that has the same body as the original function.
     * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
     * @param thisArg The object to be used as the this object.
     * @param args Arguments to bind to the parameters of the function.
     */
    bind<T>(this: T, thisArg: ThisParameterType<T>): OmitThisParameter<T>;
    bind<T, A0, A extends any[], R>(this: (this: T, arg0: A0, ...args: A) => R, thisArg: T, arg0: A0): (...args: A) => R;
    bind<T, A0, A1, A extends any[], R>(this: (this: T, arg0: A0, arg1: A1, ...args: A) => R, thisArg: T, arg0: A0, arg1: A1): (...args: A) => R;
    bind<T, A0, A1, A2, A extends any[], R>(this: (this: T, arg0: A0, arg1: A1, arg2: A2, ...args: A) => R, thisArg: T, arg0: A0, arg1: A1, arg2: A2): (...args: A) => R;
    bind<T, A0, A1, A2, A3, A extends any[], R>(this: (this: T, arg0: A0, arg1: A1, arg2: A2, arg3: A3, ...args: A) => R, thisArg: T, arg0: A0, arg1: A1, arg2: A2, arg3: A3): (...args: A) => R;
    bind<T, AX, R>(this: (this: T, ...args: AX[]) => R, thisArg: T, ...args: AX[]): (...args: AX[]) => R;

```